internal/storage/storage.go

```diff
@@ -0,0 +1,60 @@
+package storage
+
+import (
+	"time"
+)
+
+// Backend is an interface for defining functions which all storage providers support.
+type Backend interface {
+	Copy(file string) error
+	Prune(deadline time.Time, pruningPrefix string) (*PruneStats, error)
+	GetName() string
```

Comment:
`GetFoo` is not really a Go thing to do, usually you do `Foo()` and 'SetFoo(fooType)`, can we use `Name()` here instead?